ci/generate_lerna_config.sh

```diff
@@ -5,84 +5,65 @@ set -o nounset
 
 PACKAGES_ARRAY=($(cat lerna_packages))
 CI_CONFIG_FILE=".circleci/lerna_config.yml"
+TESTS_PATH="ci/tests"
 
 if [ ${#PACKAGES_ARRAY[@]} -eq 0 ]; then
-
-  cat <<EOF >> $CI_CONFIG_FILE
-version: 2.1
-
-jobs:
-  tests-required:
-    docker:
-      - image: circleci/node:lts
-    steps:
-      - run:
-          name: Test dependencies
-          command: |
-            echo "No packages for testing."
-            circleci-agent step halt;
-
-workflows:
-  version: 2.1
-  build_and_test:
-    jobs:
-      - tests-required
-EOF
-
+  /bin/bash $TESTS_PATH/test-not-required.sh >> $CI_CONFIG_FILE
 else
 
   echo "Packages to test:"
   echo ${PACKAGES_ARRAY[*]}
 
   printf "version: 2.1\n\njobs:\n" >> $CI_CONFIG_FILE
 
+  if [[ " ${PACKAGES_ARRAY[@]} " =~ " @uma/financial-templates-lib " ]]; then
+      /bin/bash $TESTS_PATH/test-financial-templates-lib.sh >> $CI_CONFIG_FILE
+  fi
+
+  if [[ " ${PACKAGES_ARRAY[@]} " =~ " @uma/liquidator " ]]; then
+      /bin/bash $TESTS_PATH/test-liquidator.sh >> $CI_CONFIG_FILE
+  fi
+
   for PACKAGE in "${PACKAGES_ARRAY[@]}"
     do
-      cat <<EOF >> $CI_CONFIG_FILE
-    test-${PACKAGE:5}:
-      docker:
-        - image: circleci/node:lts
-        - image: trufflesuite/ganache-cli
-          command: ganache-cli -i 1234 -l 9000000 -p 9545
-      working_directory: ~/protocol
-      resource_class: medium+
-      steps:
-        - restore_cache:
-            key: protocol-completed-build-{{ .Environment.CIRCLE_SHA1 }}
-        - run:
-            name: Run tests
-            command: |
-              ./ci/truffle_workaround.sh
-              yarn test --scope ${PACKAGE};
-EOF
+      /bin/bash $TESTS_PATH/test-packages.sh $PACKAGE >> $CI_CONFIG_FILE
   done
 
-  cat <<EOF >> $CI_CONFIG_FILE
-    tests-required:
-      docker:
-        - image: circleci/node:lts
-      steps:
-        - run:
-            name: Test dependencies
-            command: |
-              echo "All tests running successfully"
-              circleci-agent step halt;
-EOF
+  /bin/bash $TESTS_PATH/test-required.sh >> $CI_CONFIG_FILE
 
 
   printf "\n\nworkflows:\n  version: 2.1\n  build_and_test:\n    jobs:\n" >> $CI_CONFIG_FILE
 
-  for PACKAGE in "${PACKAGES_ARRAY[@]}"
+  if [[ " ${PACKAGES_ARRAY[@]} " =~ " @uma/financial-templates-lib " ]]; then
+    REMOVE=@uma/financial-templates-lib
+    PACKAGES_ARRAY=( "${PACKAGES_ARRAY[@]/$REMOVE}" )
```

Comment:
what is this magic exactly? can you explain a bit? perhaps a comment might help as well.

Reply:
First we transform the packages edited in a list, and create an array with that information. That works until the parallelism appear. Now we need execute separated tests for each tool used to test a package. In this case, we search inside the array the `financial-templates-lib` package. If exists, we remove that package from the array and add `financial-templates-lib-hardhat` and `financial-templates-lib-truffle` that make sense to you? @chrismaree